### PR TITLE
fix(fms): gs intcp altitude

### LINKED ISSUE #8810: GS INTCP Altitude Incorrect
### Aircraft Version

Development

### Build info

```json
82739a0
```

### Describe the bug

The GS INTCP altitude shown on the dest VERT REV page is incorrect since fms-v2 merge.

### Expected behavior

The correct GS intercept alt should be shown.

### Steps to reproduce

Select NZAA ILS 23L, note the GS INCTP is 2620 feet.
image
It should be 3000 feet.
image

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #8810

## Summary of Changes

Implements the FACF flag for localiser-based approaches, and corrects the glideslope intercept altitude according to ARINC 424 coding rules.

## Screenshots (if necessary)

Before:
image
After:
image

## References

ARINC 424

## Additional context

No changelog as this is an fms-v2 bug that's never been in a stable release.

Discord username (if different from GitHub):

## Testing instructions

Select some ILS approaches of your choice in a flight plan, and check the GS INTCP altitude on the destination VERT REV page against charts.

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: fbw-common/src/systems/navdata/client/backends/Msfs/Mapping.ts [hunk 1]
```diff
@@ -1187,7 +1198,30 @@ export class MsfsMapping {
     }
   }
 
-  private mapMsLegToApproachWaypointDescriptor(leg: JS_Leg): ApproachWaypointDescriptor {
+  private isLocBasedApproach(approachType?: MSApproachType): boolean {
+    // Localizer based approaches according to ARINC 424 6.4.1
+    switch (approachType) {
```
Comment (BlueberryKing) [inline fbw-common/src/systems/navdata/client/backends/Msfs/Mapping.ts:1203]: Should this include LOC B/C approaches?

**Review by BlueberryKing — APPROVED** [review]
LGTM

**Comment by pilotseyea350** [comment]
Quality Assurance Trainee Report

Discord          : piloteyesa350
Object of testing: 8811
Tier of Testing  : 1
Date             : 20/08/2024

Testing Process: 
- Spawn at LHBP
- Power up
- ADIRS to NAV
- On INIT, set from/to to LHBP/EDDF
- On F-PLN, Choose ILS07LZ
- Check VERT DEV at EDDF07L
- Crosscheck G/S INTCP ALT == 5000 with EDDF chart 11-1
-On F-PLN, Choose ILS25L
- Check VERT DEV at EDDF25L
- Crosscheck G/S INTCP ALT == 4000 with EDDF chart 11-6
- On INIT, set from/to to LHBP/KSFO
- On F-PLN, Choose ILS28R
- Check VERT DEV at KSFO28R
- Crosscheck G/S INTCP ALT == 1800 with KSFO chart 11-3
-On F-PLN, Choose ILS19L
- Check VERT DEV at KSFO19L
- Crosscheck G/S INTCP ALT == 2800 with KSFO chart 11-1
Negatives:       
(Any issues, doubts. - type N/A if none are found)

Testing Results: 
Passed

Conclusions: 
Test passed, aircraft works as expected.

**Comment by nebj00la** [comment]
Quality Assurance Trainee Report

Discord Username  : nebj00la
Object of testing : #8811
Aircraft          : A32NX
Tier of Testing   : 2
Date              : 20/08/2024

Testing Process: 

1. Configured approach for KBOS runway 27.
2. Confirmed G/S INTCP value of 1700 matches chart for specified runway.

Testing Results: 
Passed

Media:

**Pull request merged**